framework/common/groovyScripts/CommonServices.groovy

```diff
@@ -350,107 +335,112 @@ def createKeywordThesaurus() {
-    parties = from('PartyAcctgPreference').where('enableAccounting','Y').queryIterator()
-    while (party = parties.next()){
+    parties = from('PartyAcctgPreference').where('enableAccounting', 'Y').queryIterator()
+    while (parties.hasNext()) {
+        party = parties.next()
         parameters.organizationPartyId = party.partyId
         createCustomTimePeriod = from('SystemProperty')
-            .where('systemResourceId','general', 'systemPropertyId','CustomTimePeriod.create').queryOne()
-        if ("Y".equals(createCustomTimePeriod.systemPropertyValue)) {
+                .where('systemResourceId', 'general', 'systemPropertyId', 'CustomTimePeriod.create').queryOne()
+        if (createCustomTimePeriod.systemPropertyValue == 'Y') {
             // get list of CustomTypePeriod types
             applTypes = from('SystemProperty')
-                .where('systemResourceId','general', 'systemPropertyId','CustomTimePeriod.applType').queryOne()
-            List types = Arrays.asList(applTypes.systemPropertyValue.split("\\s*,\\s*"))
-            types.each{periodTypeId ->
-                Calendar periodCal = Calendar.getInstance();
-                systemPropertyId = "CustomTimePeriod." + periodTypeId + ".intermediate"
+                    .where('systemResourceId', 'general', 'systemPropertyId', 'CustomTimePeriod.applType').queryOne()
+            List types = Arrays.asList(applTypes.systemPropertyValue.split('\\s*,\\s*'))
+            types.each { periodTypeId ->
+                Calendar periodCal = Calendar.getInstance()
+                systemPropertyId = 'CustomTimePeriod.' + periodTypeId + '.intermediate'
                 applTypeInter = from('SystemProperty')
-                    .where('systemResourceId','general', 'systemPropertyId',systemPropertyId).queryOne()
+                    .where('systemResourceId', 'general', 'systemPropertyId', systemPropertyId).queryOne()
                 if (applTypeInter) {
                     intermediate = applTypeInter.systemPropertyValue
                 }
                 // get grain for application type
-                systemPropertyId = "CustomTimePeriod." + periodTypeId + ".grain"
+                systemPropertyId = 'CustomTimePeriod.' + periodTypeId + '.grain'
                 applTypeGrain = from('SystemProperty')
-                    .where('systemResourceId','general', 'systemPropertyId',systemPropertyId).queryOne()
+                        .where('systemResourceId', 'general', 'systemPropertyId', systemPropertyId).queryOne()
                 if (applTypeGrain) {
                     grain = applTypeGrain.systemPropertyValue
-                    if ("MONTH".equals(grain)) {
+                    if (grain == 'MONTH') {
                         periodCal.add(Calendar.MONTH, 1)
-                        monthName = new SimpleDateFormat("MMM").format(periodCal.getTime())
+                        monthName = new SimpleDateFormat('MMM', locale).format(periodCal.getTime())
```

Comment:
Specifying the locale is a behaviour change. I agree that the new code is probably the right thing to do, but I think the change should be made on a different PR with a corresponding bug report.

Perhaps I'm being too pedantic. What do others think?

Reply:
Even if i agree with you, this edit is to comply the codenarc SimpleDateFormatMissingLocale rule.

So i guess we should keep it


Be sure to specify a Locale when creating a new instance of SimpleDateFormat; the class is locale-sensitive. If you instantiate SimpleDateFormat without a Locale parameter, it will format the date and time according to the default Locale. Both the pattern and the Locale determine the format. For the same pattern, SimpleDateFormat may format a date and time differently if the Locale varies

Reply:
I agree with the change if since it is needed to satisfy the codenarc rule and doesn't appear to affect the desired outcome for the user. Thanks